### ISSUE TITLE
Issues with agent_indicator_v0

### ISSUE THREAD

**Issue by nmerrill67** [PettingZoo-Team/SuperSuit#91]
Hello, I am using the atari `pong_v1` environment with the following preprocessing steps

```
    env = pong_v1.env(obs_type="grayscale_image", num_players=2,
            full_action_space=False, auto_rom_install_path="./")

    # as per openai baseline's MaxAndSKip wrapper, maxes over the last 2 frames
    # to deal with frame flickering
    env = supersuit.max_observation_v0(env, 2)

    # repeat_action_probability is set to 0.25 to introduce non-determinism to the system
    env = supersuit.sticky_actions_v0(env, repeat_action_probability=0.25)

    # skip frames for faster processing and less control
    # to be compatable with gym, use frame_skip(env, (2,5))
    env = supersuit.frame_skip_v0(env, 4)

    # downscale observation for faster processing
    env = supersuit.resize_v0(env, 84, 84)

    # allow agent to see everything on the screen despite Atari's flickering screen problem
    env = supersuit.frame_stack_v1(env, 4)

    # Include one hot vector with the agent ID in the input
    env = supersuit.agent_indicator_v0(env, type_only=True)

```

I have found two issues with `agent_indicator_v0` when being used with SB3. First, the extra channel for the one-hot vector is filled with 1's although, with image input, SB3 will [divide the input by 255](https://github.com/DLR-RM/stable-baselines3/blob/180a2e3832c12e1742545d5dc40fc7e0c8bc8009/stable_baselines3/common/preprocessing.py#L103), resulting in the one hot vector containing 0 and 1/255 now for the agent indicator.

This however, is not the main issue. The main problem is that SB3 does not detect the input as an image anymore and thus does not transpose the tensor. The first main failure point stems from the fact that `agent_indicator_v0` makes the tiled/concatenated one hot vector 1's and 0's causing [this check in SB3](https://github.com/DLR-RM/stable-baselines3/blob/180a2e3832c12e1742545d5dc40fc7e0c8bc8009/stable_baselines3/common/preprocessing.py#L49) to fail and skip transposing the tensor. So I believe that the one hot vector should be filled with 255 instead of 1 to fix this issue and the first problem that SB3 will divide by 255 anyway.

Now, assuming that [this line](https://github.com/PettingZoo-Team/SuperSuit/blob/26ac9f0228c7969a87bd20d07de06f13eb5a7401/supersuit/utils/agent_indicator.py#L19) is changed to `... = 255 * np.ones(...)` there is another issue. SB3 also [checks that the number of channels is 1, 3, or 4](https://github.com/DLR-RM/stable-baselines3/blob/180a2e3832c12e1742545d5dc40fc7e0c8bc8009/stable_baselines3/common/preprocessing.py#L61) to consider the tensor an image that needs transposing, but with my current setup there will be 6 channels, so this will fail even if the first issue is addressed.

So I am wondering if  `agent_indicator_v0` should just perform the divide by 255 and transpose to put channels first to fix this issue? The same could be said for `frame_stack_v1` if using anything besides 1, 3 or 4. Otherwise this is more like an issue with SB3, which I can post there, because I believe that in  most cases, 3D input should be transposed to put channels first for PyTorch whether it is an "image" or not. For example, a grayscale and depth image (2 channel) would currently not get transposed.

In short, I am just going to hack this together for now to make the code work, but I would like some input from the devs as to what course to take from here.

**Comment by benblack769** [comment]
Hi,

All of the above problem can be solved with supersuit.

To solve the problem where the agent indication is on a different scale from the image pixel values, you can normalize the observation:

```
env = supersuit.normalize_v0(env)
```

This works since the high and low of the observation space is different for the image channels and the agent indication channels.

To solve the problem with transposition, you can use supersuit to transpose the image instead of relying on SB3 to do it for you:


```
env = supersuit.observation_lambda_v0(env,lambda obs: np.transpose(obs, axes=(2,0,1)))
```

**Comment by nmerrill67** [comment]
This is a good solution thank you.
